Python interfaces to LArSoft [5/5]

PetrilloAtWork opened the pull request:
## Description

Python interface code has been consolidated from `icarusalg`/`sbndcode` into `sbnalg`, and at the same time it has been updated.
This pull request conforms the Python interface code after those changes, and it updates its documentation. Several files have been removed as they are now provided with the same name and interfaces in `sbnalg`.

This pull request depends on SBNSoftware/sbnalg#1 and must be merged at the same time as that one. Note that the branch in this PR was based not on `develop` but rather on `feature/mdeltutt_python_services`, subject of #640: this PR should be merged after that one too (this one effective contains PR #640).

Suggested reviewers:
* @marcodeltutto as expert in Python geometry interface
* @absolution1 as SBND software coordinator

bear-is-asleep commented:
@marcodeltutto and @dombarker30 can you review this? I think the upstream changes are in tagged releases so this is ready to merge.

marcodeltutto commented:
Sorry, I missed this one. This is good to me, tested.

bear-is-asleep commented:
trigger build LArSoft/lar*@LARSOFT_SUITE_v10_04_06 SBNSoftware/sbncode@v10_04_06_p01 SBNSoftware/sbnanaobj@bug/coackley_caf_Mar25 SBNSoftware/sbnobj@v10_00_07

FNALbuild commented:
:x: CI build for SBND Failed at phase build SBND on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/16694&builds=)

:rotating_light: For more details about the failed phase, check the [build SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/16694&platform=Linux%20slf7&phase=build&buildtype=slf7%20c14:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/25782&builds=)

FNALbuild commented:
:x: CI build for SBND Failed at phase ci_tests SBND on slf7 for e26:prof - ignored failure for unit_test - ignored warnings for build -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/16691&builds=)

:rotating_light: For more details about the failed phase, check the [ci_tests SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/16691&platform=Linux%20slf7&phase=ci_tests&buildtype=slf7%20e26:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/25781&builds=)

bear-is-asleep commented:
Hi @PetrilloAtWork , there are a few CI failures and a conflict to be resolved. Let me know if the trigger revisions need to be changed to different versions. You can see the versions in the command `trigger build` above.

## CI errors

```9:   The following imported targets are referenced, but are missing:
10:   sbnanaobj::StandardRecord sbnanaobj::StandardRecordFlat```

```14:   File "/scratch/workspace/sbnd_ci/label_exp/swarm/label_exp2/swarm/SBND/srcs/sbndcode/test/python/sbnd_services_test.py", line 2, in <module>
15:     import SBNDservices
16: ModuleNotFoundError: No module named 'SBNDservices'
17: <end of output>
18: Test time =   0.07 sec
```

PetrilloAtWork commented:
I have updated the branch to `develop` and fixed the issue with the Gallery test build (that is the failing integration test).
I haven't used the special branches mentioned in the C.I. build trigger above, just release versions plus `sbnobj` `develop`. I ran only the Gallery build test.

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/25804&builds=)

FNALbuild commented:
:x: CI build for SBND Failed at phase build SBND on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/16713&builds=)

:rotating_light: For more details about the failed phase, check the [build SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/16713&platform=Linux%20slf7&phase=build&buildtype=slf7%20c14:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/25804&builds=)

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/25803&builds=)

FNALbuild commented:
:warning: CI build for SBND Warning at phase ci_tests SBND on slf7 for e26:prof - ignored warnings for build -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/16714&builds=)

:rotating_light: For more details about the warning phase, check the [ci_tests SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/16714&platform=Linux%20slf7&phase=ci_tests&buildtype=slf7%20e26:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/25803&builds=)

bear-is-asleep commented:
Approved
